gravity-forms/gw-gravity-forms-rename-uploaded-files.php

```diff
@@ -241,32 +241,28 @@ function is_path( $filename ) {
 		return strpos( $filename, '/' ) !== false;
 	}
 
-	function get_template_value( $template, $file, $entry ) {
-
+	function get_renamed_filepath( $template, $file, $entry ) {
 		$info = pathinfo( $file );
 
-		if ( strpos( $template, '/' ) === 0 ) {
-			$dir      = wp_upload_dir();
-			$template = $dir['basedir'] . $template;
-		} else {
-			$template = $info['dirname'] . '/' . $template;
-		}
-
 		// replace our custom "{filename}" psuedo-merge-tag
-		$value = str_replace( '{filename}', $info['filename'], $template );
+		$filename = str_replace( '{filename}', $info['filename'], $template );
 
 		// replace merge tags
-		$form  = GFAPI::get_form( $entry['form_id'] );
-		$value = GFCommon::replace_variables( $value, $form, $entry, false, true, false, 'text' );
+		$form     = GFAPI::get_form( $entry['form_id'] );
+		$filename = GFCommon::replace_variables( $template, $form, $entry, false, true, false, 'text' );
+		// make sure filename is "clean". This includes removing any user inputted items such as "../", "/usr/bin" etc
+		$filename = sanitize_file_name( $filename );
```

Comment:
It was really annoying me that `$this->clean()` was just aliasing another function :laughing:

Reply:
Haha, I feel ya. The reason this it was pulled out into its own function is to allow folks to extend the class and provide their own cleaning function depending on their needs. We'll need to preserve it for that reason and backwards compat.

Reply:
Got it! Will revert.

I am curious, given that this is a snippet and folks are manually copying this into their site somewhere, why would someone extend the class to override the `clean` method rather than just modifying the code where the string cleaning happens?

It seems like an odd pattern, and I'd love to understand the benefits.

@spivurno

Reply:
Good question! 

Since we continue to make updates/fixes to the snippet, altering the class directly makes it more difficult for them to copy-and-paste the latest version. By extending the class and leaving the original class intact, they can safely update it in the future while preserving their own changes.